corehq/util/couch_doc_processor.py

```diff
@@ -367,3 +382,47 @@ def _processing_complete(self):
         ))
         if self.skipped:
             print(DOCS_SKIPPED_WARNING.format(self.skipped))
+
+
+class BulkDocProcessor(CouchDocumentProcessor):
+    """Process couch docs in batches
+
+    The bulk doc processor will send a batch of documents to the document
+    processor. If the processor does not respond with True then
+    the iteration is halted. Restarting the iteration will start by
+    re-sending the previous chunk to the processor
```

Comment:
reopened here https://github.com/dimagi/commcare-hq/pull/12299 - I'll add a test case for that